add list of goods solution

vk-workshop opened the pull request:
[DEMO LINK](https://vk-workshop.github.io/react_list-of-goods-js/)

nazarmatsevych commented on src/App.jsx:
```diff
@@ -14,45 +15,107 @@ export const goodsFromServer = [
+          Reverse
+        </button>
+
+        {reset && (
```
```suggestion
        {methodSort || reverse && (
```

and you can remove reverse state

nazarmatsevych commented on src/App.jsx:
```diff
@@ -14,45 +15,107 @@ export const goodsFromServer = [
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setReset(false);
+              setMethodSort('');
+              setReverse(false);
+            }}
```
Move to separate function to make JSX cleaner, fix all such cases

nazarmatsevych commented on src/App.jsx:
```diff
@@ -14,45 +15,107 @@ export const goodsFromServer = [
+        <button
+          type="button"
+          className={methodSort === LENGTH
+            ? 'button is-success'
+            : 'button is-success is-light'}
```
YOu can use classnames library to avoid code duplication

nazarmatsevych reviewed (changes requested):
Good work, please check the comments below

vk-workshop commented:
- deleted redundal `reset`
- changed to `methodSort || reverse`
- created separate sortAndReverse function to make JSX cleaner
- added  classnames library

etojeDenys commented on src/App.jsx:
```diff
@@ -14,45 +16,107 @@ export const goodsFromServer = [
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              sortAndReverse();
+            }}
```
```suggestion
            onClick={sortAndReverse}
```
we can simplify it

etojeDenys reviewed (approved):
good job
